app/controllers/admin/dashboard_controller.rb

```diff
@@ -84,6 +84,13 @@ def get_number_purchases
     render text: User.find(params[:id]).purchases.where(is_approved: nil, is_purchased: true).count
   end
 
+  def generate_failed_report
+    failed_books = FailedUpdate.all.map &:book
+    send_data Book.to_csv(failed_books),
+      type:         "text/csv; charset=iso-8859-1; header=present",
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Unnecessary spacing detected.